cli/compose/convert/service.go

```diff
@@ -217,13 +217,17 @@ func convertServiceSecrets(
 		if gid == "" {
 			gid = "0"
 		}
+		mode := secret.Mode
+		if mode == 0 {
```

Comment:
I agree with @mstanleyjones .

Probably the type of `Mode` needs to be `*uint32` rather than `uint32`
https://github.com/docker/docker/blob/60c1eaf8f08f904248cbfa0d07d850025a35e6d0/cli/compose/types/types.go#L232